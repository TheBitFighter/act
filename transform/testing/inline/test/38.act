function sign_extend (int<8> in) : int<12>
{
   int<12> out;
   chp{
      out:=0;
      self := in{7}=0 ? out|in  :  (~out<<8)|in 
   }
}


defproc test(chan!(int<12>) OUT)
{
   int<8> c;
   int<12> o;

   chp{
      c:=140;   
      *[ c> 120 ->   o:=sign_extend(c); OUT!o ]
   }
}

test t;

