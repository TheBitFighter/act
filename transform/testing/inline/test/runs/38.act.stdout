defproc test (chan(int<12>)! OUT);
function sign_extend (int<8> in) : int<12>;

defproc test (chan(int<12>)! OUT)
{

/* instances */
int<12> o;
int<8> c;

/* connections */
chp {
c:=0x8c;*[c>0x78 -> o:=int(int(int(c,8)>>0x7&1,1)=0x0 ? int(0x0,12)|int(c,8) : ~int(0x0,12)<<0x8|int(c,8),12);OUT!o]
}
}

function sign_extend (int<8> in) : int<12>
{

/* instances */
int<12> out;

/* connections */
chp {
out:=0x0;self:=in{0x7}=0x0 ? out|in : ~out<<0x8|in
}
}


/* instances */
test t;

/* connections */
